src/sources/socket/tcp.rs

```diff
@@ -33,37 +33,50 @@ pub struct TcpConfig {
     /// The maximum buffer size, in bytes, of incoming messages.
     ///
     /// Messages larger than this are truncated.
+    ///
+    /// This option is deprecated. Configure `max_length` on the framing config instead.
+    #[configurable(deprecated)]
```

Comment:
In socket/mod.rs , the parsing of this configuration shows that it is deprecated, but the old docs don't mention that, so this is a correction over the old docs.
Also I elected not to give a default/change to non-option because it is deprecated and the "default" get's applied in a more contrived way so this seemed reasonable to keep as-is.

Reply:
👍 as we go through these I also wonder if it's been deprecated long enough to just remove...

Reply:
In this case the changes were made only in April of last year via  #12407

Reply:
jk... that code was modified last year, but it looks like the specification that framing max len should be used was already there...

This area seems a bit touchy. There is a CI failure I'm investigating now, and it may turn our that I need to modify this.

Reply:
Pfft. Yeah, something odd going on here. I am not going to rock the boat because this is a docs change. I'll add a TODO as something to look into. I kept the deprecated flag but added a default , and reverted the Default of the config structs to use the serde crate's default since without that it broke some unit test. Which makes me think it could regress behavior for a user.

Reply:
👍 yeah, definitely want to avoid regressions here - and another case where some deprecation tracking wouldn't go amiss.

I don't think we need to pull in additional work here (even though both of us love to), so punting the actual removal is fine. A TODO seems the right choice.